api/server.js

```diff
@@ -0,0 +1,122 @@
+require('dotenv').config();
+
+const express = require('express');
+const helmet = require('helmet');
+const knex = require('knex');
+const bcrypt = require('bcryptjs'); // added
+const jwt = require('jsonwebtoken');
+const cors = require('cors');
+const knexConfig = require('../knexfile.js');
+
+const server = express();
+
+const db = knex(knexConfig.development);
+
+server.use(cors());
+server.use(helmet());
+server.use(express.json());
+
+server.get('/', (req, res) => {
+  res.send('Working!');
+});
+
+const responseStatus = {
+  success: 200,
+  postCreated: 201,
+  badRequest: 400,
+  unauthorised: 401,
+  forbidden: 403,
+  notFound: 404,
+  serverError: 500
+};
+
+server.post('/api/register', async (req, res) => {
+  try {
+    const userInfo = req.body;
+    const hash = bcrypt.hashSync(userInfo.password, 12);
+    userInfo.password = hash;
+    const ids = await db('users').insert(userInfo);
+    res.status(responseStatus.postCreated).json(ids);
+  } catch (error) {
+    res.status(responseStatus.serverError).json(error);
+  }
+});
+
+function generateToken(user) {
+  const payload = {
+    username: user.username,
```

Comment:
Pretty sure you could just destructure the param on line 45 `user` => `{username}`, then you'd be able to use it on this line 47 just like `username,`